src/django/api/extended_fields.py

```diff
@@ -12,6 +12,19 @@ def create_extendedfield(field, field_value, item, contributor):
     if field_value is not None and field_value != "":
         if field == ExtendedField.NUMBER_OF_WORKERS:
             field_value = extract_range_value(field_value)
+        elif field == ExtendedField.PARENT_COMPANY:
+            matches = Contributor.objects.filter_by_name(field_value)
+            if matches.exists():
+                field_value = {
+                    'raw_value': field_value,
+                    'contributor_name': matches[0].name,
+                    'contributor_id': matches[0].id
+                }
+            else:
+                field_value = {
+                    'raw_value': field_value,
+                    'name': field_value
```

Comment:
I like using different values here (contributor_name and _id vs just name) to indicate whether we've matched it directly to a contributor or not. If we want to make the list items links in the future, we will be able to do that pretty easily.